datafusion/common/src/types/logical.rs

```diff
@@ -0,0 +1,41 @@
+use core::fmt;
+use std::{cmp::Ordering, hash::Hash, sync::Arc};
+
+use super::NativeType;
+
+/// A reference counted [`LogicalType`]
+pub type LogicalTypeRef = Arc<dyn LogicalType>;
+
+pub trait LogicalType: fmt::Debug {
+    fn native(&self) -> &NativeType;
+    fn name(&self) -> Option<&str>;
```

Comment:
As @findepi proposed (https://github.com/apache/datafusion/issues/11513#issuecomment-2401814374) maybe a more structured approach is needed.

Reply:
A more structured approach will be needed if we want to reasonably serialize generic types.